### DISCUSSION TITLE
Why doesn't the MockV3Aggregator contract implement the AggregatorV3Interface?

### DISCUSSION THREAD

**Discussion by RamanSB** [Cyfrin/foundry-full-course-cu#1556] [Q&A]
In the FundMe lesson, we are told that the priceFeedAddresses we use on existing networks will not work in Anvil as those priceFeedAddresses do not exist in Anvil. Hence we must create a MockV3Aggregator contract, why is that this contract does not implement the AggregatorV3Interface?

One would've expected to see:

contract MockV3Aggregator is AggregatorV3Interface {
      // Ensure relevant methods are implement...
 }

**Comment by EngrPips** [comment]
Hello @RamanSB , Implementing an interface is not always a strict enforcement on any contract and whenever we are creating a mock that we want to work with on a local chain such as anvil we tend to be more loose with the standard of the contract as we sole purposely need them for testing, I hope i make it make sense to you lol.

Looking forward to hear back from you .

**Reply by RamanSB** [reply]
I would've assumed we would want the mock contract to reflect or adhere to the standard / interface of the contract we are mocking.

**Reply by EngrPips** [reply]
I mean you can make your own Mock contract better if you feel the Mock doesn't fit the standard you would like them to fit but all i am saying is their is no strict standard for Mock contracts.
